File: net/YapayGatewayLib/communication/RestV3.cs
using System;
using YapayGatewayLib.Model;
using System.Net.Http;
using System.Text;

namespace YapayGatewayLib.Communication
{
    public class RestV3
    {
        public string TransactionAuthorize(Credential credential, Transaction transaction)
        {
            return null;
        }

        public string TransactionQuery(Credential credential, string storeCode, long transactionNumber)
        {
            return null;
        }

        public string TransactionCapture(Credential credential, string storeCode, long transactionNumber, long value)
        {
            return null;
        }

        public string TransactionCancel(Credential credential, string storeCode, long transactionNumber, long value)
        {
            return null;
        }

        public string OneClickRegister(Credential credential, OneClickRegisterData registerData)
        {
            return null;
        }

        public string OneClickQuery(Credential credential, string token)
        {
            return null;
        }

        public string OneClickRegisterUpdate(Credential credential, string token, OneClickRegisterData registerData)
        {
            return null;
        }

        public string OneClickAuthorize(Credential credential, string token, Transaction transaction)
        {
            return null;
        }

        public string RecurringPaymentRegister(Credential credential, RecurringPayment recurringPayment)
        {
            return null;
        }

        public string RecurringPaymentQuery(Credential credential, string storeCode, long recurringPaymentNumber)
        {
            return null;
        }

        public string RecurringPaymentCancel(Credential credential, string storeCode, long recurringPaymentNumber)
        {
            return null;
        }

        private HttpClient createHttpClient(Credential credential)
        {
            var byteArray = Encoding.ASCII.GetBytes(credential.user + ":" + credential.password);
            HttpClient client = new HttpClient();
            client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
            return client;
        }
    }
}

File: php/yapay-gw-lib/lib/communication/rest_v3.php
<?php 
	
	class RestV3 {

		private $baseUrl;

		function __construct($baseUrl)
		{
			$this->baseUrl = $baseUrl;
		}
		
		public function transactionAuthorize($credential, $transaction){
			$curl = $this->curlInit("POST", $this->baseUrl."/api/v3/transacao", $transaction, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function transactionQuery($credential, $storeCode, $transactionNumber){
			$curl = $this->curlInit("GET", $this->baseUrl."/api/v3/transacao/".$storeCode."/".$transactionNumber, null, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function transactionCapture($credential, $storeCode, $transactionNumber, $value){
			$curl = $this->curlInit("PUT", $this->baseUrl."/api/v3/transacao/".$storeCode."/".$transactionNumber, null, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function transactionCancel($credential, $storeCode, $transactionNumber, $value){
			$curl = $this->curlInit("PUT", $this->baseUrl."/api/v3/transacao/".$storeCode."/".$transactionNumber, null, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function oneclickRegister($credential, $oneclickRegisterData){
			$curl = $this->curlInit("POST", $this->baseUrl."/api/v3/oneclick/", $oneclickRegisterData, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function oneclickQuery($credential, $token){
			$curl = $this->curlInit("GET", $this->baseUrl."/api/v3/oneclick/".$token, null, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function oneclickRegisterUpdate($credential, $token, $oneclickRegisterData){
			$curl = $this->curlInit("PUT", $this->baseUrl."/api/v3/oneclick/".$token."/alterar", $oneclickRegisterData, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function oneclickAuthorize($credential, $token, $transaction){
			$curl = $this->curlInit("POST", $this->baseUrl."/api/v3/oneclick/".$token."/autorizar", $transaction, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function recurringPaymentRegister($credential, $recurringPaymentData){
			$curl = $this->curlInit("POST", $this->baseUrl."/api/v3/recorrencia", $recurringPaymentData, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function recurringPaymentQuery($credential, $storeCode, $recurringPaymentNumber){
			$curl = $this->curlInit("GET", $this->baseUrl."/api/v3/recorrencia/".$storeCode."/".$recurringPaymentNumber, null, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		public function recurringPaymentCancel($credential, $storeCode, $recurringPaymentNumber){
			$curl = $this->curlInit("PUT", $this->baseUrl."/api/v3/recorrencia/".$storeCode."/".$recurringPaymentNumber, null, $credential);
			$response = curl_exec($curl);
			curl_close($curl);
			return $response;
		}

		private function curlInit($requestType, $url, $data, $credential){
			$ch = curl_init($url);
			curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
			curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $requestType);
			curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($data));
			curl_setopt($ch, CURLOPT_USERPWD, $credential->user . ":" . $credential->password);
			curl_setopt($ch, CURLOPT_HTTPHEADER, array("Content-Type: application/json"));
			return $ch;
		}

	}

 ?>